### ISSUE TITLE
Feature Request: Additional meta for notes

### ISSUE THREAD

**Issue by jinjie** [fabiospampinato/notable#67]
It will be nice to have additional metadata such as 

- Note created
- Note last edited
- Note location

It can come in handy when you want to do quick notes during meetings. I can easily refer to the date created and the location where the note was taken so I can recall back when and where did I jot this note. Pretty cool feature, nice to have.

**Comment by fabiospampinato** [comment]
Creation date and last edited date could be added, but they would kind of be out of sync if you update/create your notes outside of Notable, maybe there could be a normalization step for this at startup, but it's a bit less trivial than it sounds at first.

Adding other metadata such as location would require a why of editing the front matter in Notable, maybe there could be an additional button in the toolbar for additional metadata 🤔

**Comment by mvakili** [comment]
creation and modification date can be handled by the operating system, we don't need to save them as metadata

**Comment by mystal** [comment]
I sync my notes using Dropbox and haven't looked into how it handles create/modification times. It sounds like you're saying Dropbox will potentially give wrong values for these? If so, having the created/modified timestamp metadata sounds very useful.

I was once working on a similar note app that stored files in Markdown with metadata in frontmatter. I gave each note tracked by the app a UUID. If Notable added a UUID to the frontmatter, it could keep track of notes across file renames or edits outside of itself. Any thoughts?

**Comment by fabiospampinato** [comment]
@mystal currently creation/modification dates aren't stored, but are inferred from the filesystem. This obviously doesn't work but I'll fix this in the next few releases. There's no need for a UUID as each note's metadata is written alongside the note's content in the same file.

**Comment by fabiospampinato** [comment]
As of https://github.com/fabiospampinato/notable/commit/3641862488ae869b41bdbf57f611666f9311f811 creation and modification times are now properly preserved. 

Support for editing custom metadata will have to wait some more time (you can already add custom metadata by editing it directly though).

**Comment by mikaelmello** [comment]
Coming from https://github.com/notable/notable/issues/538#issuecomment-475890206

> Ok, but I'm struggling to understand what the actual advantage would be here. As a mental model I'm imagining this as if there was a line that divided the note in 2 sections, the first section contains only the title and the second section contains the content. What advantages does this kind of division bring to your use case?

I think I'm not making myself quite clear:

The note title/filename will not appear the in the editor/display at all. It is something that is used to display the notes on the sidebar and to name the file. _If the user decides to use a custom title_

By default, the title would be inferred from the title and nothing would change from the current approach.

What I'm talking about is a possibility to add a custom title for notes. There are notes where the title is different from the content. For example:

> I want to write a proposal in my note and I'd like to easily find my note by looking for something such as "Proposal for Organization X - Program Y 2019" (1).
> 
> Well, my note is a proposal and I'd like the main header to be the name of the proposal: "Improving X's security" (2), not the sentence (1). This proposal is going to be viewed by other people and they all already know it's a proposal for them, they don't need to read (1) but (2).
> 
> The only usefulness for (1) is to organize myself, being it to manually search for the file in the filesystem or via Notable's sidebar, where notes are listed.
>
> (1) would not appear anywhere inside the note.

If I want the note's header and the note's title to be the same, I can just ignore this feature and move on with my life.

The way I see it being implemented is just an additional button here: 
![image](https://user-images.githubusercontent.com/17488630/54869794-73776a00-4d7c-11e9-93a0-c9f8f4b5de19.png)

You click on it, you set or remove a title, Notable acts accordingly.

**Comment by fabiospampinato** [comment]
@mikaelmello Ok I think we are on the same page now. I'm not sure _why_ someone would not want to have a note's title there immediately editable along with the content, but I see the need for better editable metadata in general. 

I've been thinking about this for a while actually, I think at some point the "Open in Default App" button will be replaced with an "Info" button, that let's you see some additional metadata like creation and modification date, let's you set a custom, fixed, title and let's you add additional metadata, which could perhaps be useful for search in the future #293.

**Comment by mikaelmello** [comment]
> I'm not sure why someone would not want to have a note's title there immediately editable along with the content

From my experience I almost never edit titles once they are set. The contents are updated many more times

**Comment by neighthan** [comment]
I'm considering migrating to Notable from just editing my notes in VS Code, but my notes already have front matter with my own metadata (including tags). I would find Notable much better if there was an option to show the metadata directly when editing the file.

**Comment by fabiospampinato** [comment]
@neighthan Maybe we could add an option for that in the future, but I think a better approach would be to show a separate UI component for editing the metadata 🤔 Like you might not want to learn how YAML's syntax works, and using a date-picker may be less error-prone than editing a date field directly.

**Comment by shah** [comment]
Agreed @fabiospampinato - I also found Notable as a useful editor to compose Hugo markdown files. The only thing missing is the custom meta data editor you reference but even @neighthan idea of just showing and saving the markdown would suffice for now.
